lib/toggles/constant_lookup.rb

```diff
@@ -8,35 +8,42 @@ def initialize(sym)
     end
   end
 
-
   # Return a tree walker that translates Module#const_missing(sym) into the next child node
   #
   # So Features::Cat::BearDog walks as:
   # * next_features = Feature.features # root
   # * const_missing(:Cat) => next_features = next_features['cat']
   # * const_missing(:BearDog) => next_features['bear_dog']
+  # * const_missing(:CN22) => next_features['c_n_22']
   #
   # Defined at Toggles.features_dir + "/cat/bear_dog.yaml"
   #
   # @raise [Error] if constant cannot be resolved
   def self.from(features, path)
     Class.new {
       class << self
-        attr_accessor :features
-        attr_accessor :path
+        attr_accessor :features, :path
 
         def const_missing(sym)
-          subtree_or_feature = features.fetch(
-            # translate class name into path part i.e :BearDog #=> 'bear_dog'
-            sym.to_s.gsub(/([a-z])([A-Z])/) { |s| s.chars.join('_') }.downcase.to_sym,
-          )
+          # translate class name into path part i.e :BearDog #=> 'bear_dog'
+          key =
+            sym
+            .to_s
+            .gsub(/([A-Z]{2,})/) { |s| s.chars.join('_') }
+            .gsub(/([a-z])([A-Z])/) { |s| s.chars.join('_') }
+            .downcase
+            .gsub(/([a-z])([0-9])/) { |s| s.chars.join('_') }
+            .to_sym
```

Comment:
This doesn't look like it'll work for strings with a number directly before a capital letter, e.g. `"Allow2DayService"`, it translates to `:allow_2day_service`, I'm thinking it should be `:allow_2_day_service` based on the CN22 example, right? Unless you didn't want the separation after the number? It looks like we'll just need another `gsub` with the converse, digit before letter, right?

Reply:
hmm, ill add another test case.